provider/aci.go

```diff
@@ -668,16 +670,38 @@ func (p *ACIProvider) CreatePod(ctx context.Context, pod *v1.Pod) error {
 	containerGroup.RestartPolicy = aci.ContainerGroupRestartPolicy(pod.Spec.RestartPolicy)
 	containerGroup.ContainerGroupProperties.OsType = aci.OperatingSystemTypes(p.operatingSystem)
 
+	masterURI := os.Getenv("MASTER_URI")
+	t := regexp.MustCompile(`[:/]`)
+	masterURISplit := t.Split(masterURI, -1)
+	clusterFqdn := ""
+	if len(masterURISplit) > 1 {
+		clusterFqdn = masterURISplit[3]
+	}
+	cluster, err := p.aciClient.GetAKSCluster(ctx, p.resourceGroup, clusterFqdn)
+	if err != nil {
+		log.G(ctx).Infof("Error getting cluster details \n Can't use kubelet identity for container %v", err)
+	}
+
```

Comment:
As we are already planning to migrate to Azure SDK, it'd recommend to create a separate folder/package for AKS and use the AKS SDK to get the cluster directly instead of listing all clusters.